### ISSUE TITLE
Limit the number of elements to 1 in initialization in yii.gridView.js 

### ISSUE THREAD

**Issue by arogachev** [yiisoft/yii2#13233]
This is similar to #13165.

I agree with https://github.com/yiisoft/yii2/issues/13165#issuecomment-266235856 that init of 2 identical captchas can be useful. But what's the point of initialize multiple grid views, for example:

```js
$('.grid-view').yiiGridView({
    filterUrl: '/posts/index',
    filterSelector: '...' // ???
});
```

Again, same settings will be applied to all GridView widgets. Usually `filterSelector` is something like `#w0-filters-input, #w0-filters-select` (related to one specific widget). If we leave it like that, both GridView widgets can be filtered using interacting with widget `#w0`. Does not make sense for me because multiple grid views usually represent different objects (even this case is rare) and it's very weird UI / UX.

The only use case I can think of is common controls, something like `#w-common-filters input, #w-common-filters select`:

```html
<div id="w-common-filters">
  <input name="PostSearch[id]" type="text">
  <input name="PostSearch[name]" type="text">
</div>
```

I used this for testing in #13230.

But this again doesn't make much sense because `PostSearch` is related to one specific model. It makes more sense with multiple grid views representing different objects with similar properties (guides containing just `id` and `name` for example) and filters without explicit mention of search model:

```html
<div id="w-common-filters">
  <input name="id" type="text">
  <input name="name" type="text">
</div>
```

or one common search model for all grid views in this page:

```html
<div id="w-common-filters">
  <input name="CommonSearch[id]" type="text">
  <input name="CommonSearch[name]" type="text">
</div>
```

Should we leave it like that or not?

**Comment by samdark** [comment]
Can't find a use case for it as well. I guess removing such ability would be a BC break anyway, right?

**Comment by arogachev** [comment]
Yes, seems like it will be a BC break. Because of that and ambiguity I haven't fixed it during writing tests.

**Comment by SDKiller** [comment]
> But what's the point of initialize multiple grid views, for example:
>
> ```
> $('.grid-view').yiiGridView({
>    filterUrl: '/posts/index',
>    filterSelector: '...' // ???
> });
> ```

What about something like the following case:

```
$('.grid-view').yiiGridView({
    filterUrl: $(this).data('filter_url'),
});
```
